Add GitHub Actions workflow to create Sentry release AND deploy to Heroku

LearningNerd opened the pull request:
(This would be merged _instead_ of PR #14.) This workflow uses the Sentry CLI to create a new release (using the current git commit SHA), associate commits, create a new deploy, and then actually deploy it to Heroku. Right now it also updates the `SENTRY_RELEASE` environment variable on Heroku, which is used to initialize/configure Sentry so any events get linked up to the latest release.

❓What makes sense for this workflow's trigger? Should I tie it to Github releases? Merges to master? (Right now, it creates a new release every time there's a commit to the master branch.)

Note: This assumes that the linked Sentry project is already configured with the Github integration.

Required Github Secrets:
  - HEROKU_API_KEY
  - SENTRY_AUTH_TOKEN
  - SENTRY_DSN
  - SENTRY_ORG
  - SENTRY_PROJECT
  - SENTRY_DEPLOY_ENVIRONMENT

LearningNerd committed:
Create main.yml

LearningNerd committed:
Add workflow to create new Sentry release, udpate Heroku env var

LearningNerd committed:
Combine workflows into one file

LearningNerd commented:
UPDATES (thanks @manuzope !):
  - Combined into one workflow file (so this will replace PR #14 )
  - Now uses specific version of `ubuntu-18.04` instead of `ubuntu-latest`

LearningNerd committed:
Update Sentry init to use release if available

LearningNerd committed:
fix typo

scefali commented on sentry.js:
```diff
@@ -1,12 +1,18 @@
+    dsn: sentryDSN
+  };
+
+  if (sentryRelease != undefined) {
```
@LearningNerd we should avoid using non-strict equality checks like `!=`. It can produce unintended side effects. I would recommend just doing:
```
  if (sentryRelease) {
```

LearningNerd replied:
Alrighty, I'll update that line now!

LearningNerd committed:
Use if(x) instead of (x != undefined)

for sentryDSN and sentryRelease

scefali commented on .github/workflows/sentry-and-heroku-deploy.yml:
```diff
@@ -0,0 +1,40 @@
+        sentry-cli releases new -p $SENTRY_PROJECT $SENTRY_RELEASE
+        sentry-cli releases finalize $SENTRY_RELEASE
+        sentry-cli releases set-commits --auto $SENTRY_RELEASE
+        sentry-cli releases deploys $SENTRY_RELEASE new -e $SENTRY_DEPLOY_ENVIRONMENT
```
@LearningNerd This looks good to me but I recommend getting a sales engineer to review this as well, ideally Neil if he has time.

LearningNerd replied:
👍 will do :)

ndmanvar replied:
this looks good! i think finalize should go after set-commits.

LearningNerd replied:
Thanks @ndmanvar ! I switched that and fixed it up a bit.

LearningNerd committed:
Fix bug with SENTRY_RELEASE being reset

Combine run steps so that `SENTRY_RELEASE` doesn't get reset between steps. Add comments. Finalize release after associating commits.
